parsl/executors/base.py

```diff
@@ -32,6 +32,16 @@ class ParslExecutor(metaclass=ABCMeta):
               @typeguard the constructor, you'll have to use List[Any] here.
     """
 
+    # This __init__ method exists to provide a python 3.5 compatible type declaration
+    # for the `label` attribute. If/when python 3.5 is deprecated (see parsl issue #1553)
+    # this could be replaced with a python 3.6 style attribute type declaration.
+    #   label: str
+    # and __init__ removed.
+
+    @abstractmethod
+    def __init__(self) -> None:
```

Comment:
We key the executors on the label, so it would be very nice to make label a required argument once we switch to 3.6.